src/buildstream/_frontend/app.py

```diff
@@ -375,10 +375,6 @@ def init_project(
 
         project_path = os.path.join(directory, "project.conf")
 
-        if min_version is None:
```

Comment:
Here also, I'm not sure why we're breaking the behavior of `bst init` here so that it no longer initializes a project with the min-version set to the installed version, if `bst init` is not given the `--version` argument.